cmd/qat_plugin/qat_plugin.go

```diff
@@ -15,323 +15,52 @@
-	err = ioutil.WriteFile(bindDevicePath, []byte(vendorPrefix+vfdevID), 0644)
-	if err != nil {
-		return errors.Wrapf(err, "Binding to the DPDK driver failed for the device %s", id)
-	}
-	return nil
-}
-
-func isValidKerneDriver(kernelvfDriver string) bool {
-	switch kernelvfDriver {
-	case "dh895xccvf", "c6xxvf", "c3xxxvf", "d15xxvf":
-		return true
-	}
-	return false
-}
-
-func isValidDpdkDeviceDriver(dpdkDriver string) bool {
-	switch dpdkDriver {
-	case "igb_uio", "vfio-pci":
-		return true
-	}
-	return false
-}
-func isValidVfDeviceID(vfDevID string) bool {
-	switch vfDevID {
-	case "0442", "0443", "37c9", "19e3":
-		return true
-	}
-	return false
-}
-
-func (dp *devicePlugin) PostAllocate(response *pluginapi.AllocateResponse) error {
-	tempMap := make(map[string]string)
-	for _, cresp := range response.ContainerResponses {
-		counter := 0
-		for k := range cresp.Envs {
-			tempMap[strings.Join([]string{"QAT", strconv.Itoa(counter)}, "")] = cresp.Envs[k]
-			counter++
-		}
-		cresp.Envs = tempMap
-	}
-	return nil
-}
-
-func (dp *devicePlugin) scan() (deviceplugin.DeviceTree, error) {
-	devTree := deviceplugin.NewDeviceTree()
-	n := 0
-	for _, driver := range append([]string{dp.dpdkDriver}, dp.kernelVfDrivers...) {
-		files, err := ioutil.ReadDir(path.Join(dp.pciDriverDir, driver))
-		if err != nil {
-			fmt.Printf("Can't read sysfs for driver as Driver %s is not available: Skipping\n", driver)
-			continue
-		}
-
-		for _, file := range files {
-			if !strings.HasPrefix(file.Name(), "0000:") {
-				continue
-			}
-			vfdevID, err := dp.getDeviceID(file.Name())
-			if err != nil {
-				return nil, errors.Wrapf(err, "Cannot obtain deviceID for the device with PCI address: %s", file.Name())
-			}
-			if !isValidVfDeviceID(vfdevID) {
-				continue
-			}
-			n = n + 1 // increment after all junk got filtered out
-
-			if n > dp.maxDevices {
-				break
-			}
-
-			vfpciaddr := strings.TrimPrefix(file.Name(), "0000:")
-
-			// initialize newly found devices which aren't bound to DPDK driver yet
-			if driver != dp.dpdkDriver {
-				err = dp.bindDevice(vfpciaddr)
-				if err != nil {
-					return nil, err
-				}
-			}
-
-			devNodes, err := dp.getDpdkDeviceSpecs(vfpciaddr)
-			if err != nil {
-				return nil, err
-			}
-			devMounts, err := dp.getDpdkMounts(vfpciaddr)
-			if err != nil {
-				return nil, err
-			}
-			deviceName := strings.TrimSuffix(namespace, ".intel.com")
-
-			devinfo := deviceplugin.DeviceInfo{
-				State:  pluginapi.Healthy,
-				Nodes:  devNodes,
-				Mounts: devMounts,
-				Envs: map[string]string{
-					fmt.Sprintf("%s%d", strings.ToUpper(deviceName), n): file.Name(),
-				},
-			}
-
-			devTree.AddDevice("generic", vfpciaddr, devinfo)
-		}
-	}
+func main() {
+	var plugin deviceplugin.Scanner
+	var err error
 
-	return devTree, nil
-}
+	debugEnabled := flag.Bool("debug", false, "enable debug output")
+	mode := flag.String("mode", "vfio", "plugin mode which can be either vfio (default) or kernel")
```

Comment:
Can this mode name be changed from "vfio" to "dpdk"? As dpdk drivers include vfio-pci and igb-uio this would avoid confusion between vfio and vfio-pci. This would also mean changing directory and files with names "vfiodrv" to "dpdkdrv".

Reply:
Sure, renamed everything to dpdk.